transit_odp/otc/client/otc_client.py

```diff
@@ -118,6 +118,29 @@ def get_latest_variations_since(self, when: datetime) -> List[Registration]:
 
         return variations
 
+    def get_latest_variations_by_registration_code(
+        self, registration_codes: list
+    ) -> List[Registration]:
+        variations = []
+        for registration_code in registration_codes:
+            logger.info(
+                f"Requesting latest variation for registration - {registration_code} from OTC API"
+            )
+            response = self._make_request(
+                page=1, regNo=registration_code, latestVariation=True
+            )
+            variations += response.bus_search
+
+            for page in range(2, response.page.total_pages + 1):
+                response = self._make_request(
+                    page=page,
+                    regNo=registration_code,
+                    latestVariation=True,
+                )
+                variations += response.bus_search
+
+            return variations
```

Comment:
The return statement should be outside the for loop. otherwise this function will exit after the first item in the loop.